### PR TITLE
feat(console): Template grouping logic for IdPs

### PR DESCRIPTION
**Pull request by yasinmiran**
### Related Issues
(None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation
- [ ] Documentation provided. (Add links)
- [ ] Unit tests provided. (Add links if any)
- [ ] Integration tests provided. (Add links if any)

### Related PRs
- Following depends on this PR: **https://github.com/wso2/identity-apps/pull/2052**

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [x] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/identity-providers/components/wizards/authenticator-create-wizard-factory.tsx [hunk 1]
```diff
@@ -74,6 +83,16 @@ export const AuthenticatorCreateWizardFactory: FunctionComponent<AuthenticatorCr
...
+        /**
+         * Added this because {@link IdentityProviderTemplateSelectPage} currently
+         * unable to handle grouped templates properly. {@link selectedTemplate}
+         * will be a grouped identity provider with sub templates. Even though, the
+         * grouping logic implemented and is in place {@link getTemplate} method
+         * keeps failing to set the correct grouped template to this state.
+         *
+         * FIXME: As a part of https://github.com/wso2-enterprise/asgardeo-product/issues/3878
+         */
+        selectedTemplate: parentSel,
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/wizards/authenticator-create-wizard-factory.tsx:95]: Let's add a meaningful alias here.
```suggestion
        selectedTemplate: parentSel,
```

Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/wizards/authenticator-create-wizard-factory.tsx]: +1 Addressed this in commit 63790c8

**Comment by yasinmiran** [comment]
These changes will be patched and applied in https://github.com/wso2/identity-apps/pull/2052. Closing due to conflicts and code interdependencies of the targeted feature.

File: apps/console/src/extensions/extensions-manager.ts [hunk 2]
```diff
@@ -145,6 +144,14 @@ export class ExtensionsManager {
                 });
         }
 
+        // If groups exists, try to resolve the group config by lazy loading the resource etc.
```
Comment (thivi) [inline apps/console/src/extensions/extensions-manager.ts:147]: It should be `If groups exist".

File: apps/console/src/features/identity-providers/utils/identity-provider-template-management-utils.ts [hunk 3]
```diff
@@ -65,11 +63,33 @@ export class IdentityProviderTemplateManagementUtils {
...
+                            .then((groups: IdentityProviderTemplateInterface[]) => {
+                                if (sort) {
+                                    templates = IdentityProviderTemplateManagementUtils
+                                        .sortIdentityProviderTemplates(templates);
+                                    groups = IdentityProviderTemplateManagementUtils
+                                        .sortIdentityProviderTemplates(groups);
+                                }
+                                store.dispatch(setIdentityProviderTemplates(templates));
+                                store.dispatch(setIdentityProviderTemplates(groups, true));
+                                return Promise.resolve();
```
Comment (thivi) [inline apps/console/src/features/identity-providers/utils/identity-provider-template-management-utils.ts:88]: Add a new line before the return statement.

File: apps/console/src/features/identity-providers/utils/identity-provider-template-management-utils.ts [hunk 4]
```diff
@@ -241,26 +261,102 @@ export class IdentityProviderTemplateManagementUtils {
...
+
+        const groupedTemplates: IdentityProviderTemplateInterface[] = [];
+
+        return IdentityProviderTemplateManagementUtils
+            .loadLocalFileBasedIdentityProviderTemplateGroups()
+            .then((response: IdentityProviderTemplateGroupInterface[]) => {
+                templates.forEach((template: IdentityProviderTemplateInterface) => {
+                    if (!template.templateGroup) {
+                        groupedTemplates.push(template);
+                        return;
```
Comment (thivi) [inline apps/console/src/features/identity-providers/utils/identity-provider-template-management-utils.ts:281]: Add a new line before the return statement. Check the other places as well.

**Pull request closed without merging**